SignalServiceKit/src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -159,17 +251,56 @@ - (instancetype)initOutgoingMessageWithTimestamp:(uint64_t)timestamp
         return self;
     }
 
-    _messageState = TSOutgoingMessageStateAttemptingOut;
-    _sentRecipients = [NSArray new];
     _hasSyncedTranscript = NO;
     _groupMetaMessage = groupMetaMessage;
     _isVoiceMessage = isVoiceMessage;
 
     _attachmentFilenameMap = [NSMutableDictionary new];
 
+    // New outgoing messages should immediately determine their
+    // recipient list from current thread state.
+    NSMutableDictionary<NSString *, TSOutgoingMessageRecipientState *> *recipientStateMap = [NSMutableDictionary new];
+    NSArray<NSString *> *recipientIds = [self.thread recipientIdentifiers];
```

Comment:
Let's use the `thread` param rather than `self.thread` to avoid opening a sneaky potentially blocking transaction here.

Reply:
Good idea.